junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestNameFormatter.java

```diff
@@ -30,17 +30,19 @@
 		this.namePattern = namePattern;
 	}
 
-	String format(int invocationIndex, Object... arguments) {
+	String format(int invocationIndex, Arguments arguments) {
 		String pattern = prepareMessageFormatPattern(invocationIndex, arguments);
 		Object[] humanReadableArguments = makeReadable(arguments);
 		return formatSafely(pattern, humanReadableArguments);
 	}
 
-	private String prepareMessageFormatPattern(int invocationIndex, Object[] arguments) {
+	private String prepareMessageFormatPattern(int invocationIndex, Arguments arguments) {
+		// todo: shall it be split into three methods for clarity?
 		String result = namePattern.replace("{index}", String.valueOf(invocationIndex));
+		result = result.replace("{arguments.description}", arguments.description());
 		if (result.contains("{arguments}")) {
 			// @formatter:off
-			String replacement = IntStream.range(0, arguments.length)
+			String replacement = IntStream.range(0, arguments.get().length)
```

Comment:
I'd like to avoid calling `get()` multiple times since it may be costly to compute the array.

Reply:
Sure can do, although I'm not sure I understand the reasoning. `Arguments.of` captures its variable arity parameter (= a reference to an `Object[]`), and `Arguments.get` return the same reference to the same object. The current code does not perform any defensive copies because they seem to be redundant in this case (it is never OK to modify an underlying varargs parameter at call-site; regarding internal, package-private code -- we can trust it).

Reply:
I may be a bit too defensive here, but since it's an interface people might implement it and compute the array lazily. It's a bit like a `Supplier`.

Reply:
I see, thank you, I'll think how to express it better.

Just to confirm, `Arguments` contract does not allow implementers to return _different_ arrays in subsequent calls, does it? That is, I can preserve whatever the first call returns? If not, can we restrict it in the spec?

Reply:
And a couple of things to consider:
 1. Does it ever happen (if all tests are correct) that JUnit does not consume all of the arguments? I might recall a recent Run-test-by-unique-id feature :thinking: 
 2. Does it matter in terms of prospective parallel execution _where_ we actually consume the arguments?